x/signal/keeper_test.go

```diff
@@ -466,8 +466,9 @@ func setup(t *testing.T) (signal.Keeper, sdk.Context, *mockStakingKeeper) {
 			Block: 1,
 			App:   1,
 		},
-		ChainID: appconsts.TestChainID,
-	}, false, log.NewNopLogger())
+	}, false, log.NewNopLogger()).WithHeaderInfo(header.Info{
+		ChainID: appconsts.TestChainID, // TryUpgrade reads chainID from header info, not block header.
```

Comment:
is it safe to assume headerInfo will always be populated, or do we need to update the code [here](https://github.com/01builders/celestia-app/blob/a5aece3f335a08c9b23e78b958a90f31ed05f174/x/signal/keeper.go#L109-L110)

Reply:
do we require `WithHeaderInfo` here? I thought that change only really came into affect in v0.52, but I'm not 100% sure